charmcraft/commands/init.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+from datetime import date
+import logging
+import os
+from pathlib import Path
+import pwd
+import re
+
+from jinja2 import Environment, PackageLoader, StrictUndefined
+
+from charmcraft.cmdbase import BaseCommand, CommandError
+from ._utils import make_executable
+
+logger = logging.getLogger(__name__)
+
+
+class InitCommand(BaseCommand):
+    """Initialize a directory to be a charm project."""
+    name = "init"
+    help_msg = "initialize a directory to be a charm project"
+
+    def fill_parser(self, parser):
+        parser.add_argument(
+            "-d", "--directory", type=Path, default=Path("."),
+            help="the directory to initialize. Must not exist or be empty; defaults to '.'")
+        parser.add_argument(
+            "--name", type=str,
+            help="the name of the project; defaults to the directory name")
+
+        parser.add_argument(
+            "--author", type=str,
+            help="the author of the project; defaults to the directory name;"
+            " defaults to the current user's name as present in the GECOS field.")
+
+    def run(self, args):
+        args.directory = args.directory.resolve()
+        if args.directory.exists():
+            if not args.directory.is_dir():
+                raise CommandError("{} is not a directory".format(args.directory))
+            for _ in args.directory.iterdir():
+                raise CommandError("{} is not empty".format(args.directory))
+            logger.debug("Using existing project directory '%s'", args.directory)
+        else:
+            logger.debug("Creating project directory '%s'", args.directory)
+            args.directory.mkdir()
+
+        if args.author is None:
+            gecos = pwd.getpwuid(os.getuid()).pw_gecos.split(',', 1)[0]
+            if not gecos:
+                raise CommandError("Author not given, and nothing in GECOS field")
+            logger.debug("Setting author to %r from GECOS field", gecos)
```

Comment:
is GECOS useful to report? I know it doesn't mean much to me.

Reply:
I didn't knew about GECOS until now...

Reply:
if set, it's expected to be the user's name